tensorflow_cloud/run.py

```diff
@@ -0,0 +1,153 @@
+            as the base image. The version of TensorFlow and Python in that
+            case will match your local environment.
+        chief_config: Optional `MachineConfig` that represents the
+            configuration for the chief worker in a distribution cluster.
+            Defaults to 'auto'. 'auto' maps to a standard gpu config such as
+            `COMMON_MACHINE_CONFIGS.P100_1X` (8 cpu cores, 30GB memory,
+            1 Nvidia Tesla P100).
+        worker_config: Optional `MachineConfig` that represents the
+            configuration for the general workers in a distribution cluster.
+            Defaults to 'auto'. 'auto' maps to a standard gpu config such as
+            `COMMON_MACHINE_CONFIGS.P100_1X` (8 cpu cores, 30GB memory,
+            1 Nvidia Tesla P100).
+        worker_count: Optional integer that represents the number of general
+            workers in a distribution cluster. Defaults to 0. This count does
+            not include the chief worker.
+        region: Optional string. Cloud region in which to submit the
+            job. Defaults to 'us-central1' for GCP.
+        entry_point_args: Optional list of strings. Defaults to None.
+            Command line arguments to pass to the `entry_point` program.
+        stream_logs: Boolean flag which when enabled streams logs back from
+            the cloud job.
+    """
+    # If code is triggered in a cloud environment, do nothing.
+    if _is_running_remotely():
+        return
+    _set_running_remotely(True)
+
+    # Get defaults.
+    if chief_config == 'auto':
+        chief_config = machine_config.COMMON_MACHINE_CONFIGS['P100_1X']
+    if worker_config == 'auto':
+        worker_config = machine_config.COMMON_MACHINE_CONFIGS['P100_1X']
+    region = region or gcp.get_region()
+    dst_path_prefix = '/app/'
+    docker_registry = 'gcr.io/{}'.format(gcp.get_project_name())
+
+    # Run validations.
+    validate.validate(
+        entry_point, distribution_strategy, requirements_txt,
+        chief_config, worker_config, worker_count, region,
+        entry_point_args, stream_logs)
+
+    # Create the script to run (starter_script).
+    # Make the `entry_point` cloud and distribution ready.
+    startup_script = preprocess.get_startup_script(
+        entry_point, chief_config, worker_count, distribution_strategy)
+
+    # Get all the files, that we need to package, mapped to the dst location.
+    # This will include the startup script, requirements_txt, dockerfile,
+    # files in the entry_point dir.
+    dockerfile, file_map = containerize.get_file_map(
+        entry_point, startup_script, chief_config, requirements_txt,
+        dst_path_prefix, docker_base_image)
+
+    # Create a tarball with the files.
+    tarball = package.get_tarball(file_map)
+
+    # Create docker image.
+    docker_img = containerize.get_docker_image(docker_registry, tarball)
+
+    # Delete all the temporary files we created.
```

Comment:
Note that we may not necessarily have write access to create temp files. This is not necessarily a meaningful issue (since it's a rare case) , but it's something to keep in mind.

Reply:
Made 'wrapped_entry_point' script creation optional - one less deletion to worry about. I am writing some examples and detailed usages doc - will mention these in that.